lib/storage.js

```diff
@@ -0,0 +1,466 @@
+    assert.string(config.hmac.algorithm, 'config.hmac.algorithm');
+  }
+
+  // check permission against controller ID
+  const {controller} = config;
+  await brPermissionCheck(
+    actor, PERMISSIONS.DATA_HUB_CONFIG_UPDATE,
+    {resource: [controller]});
+
+  // insert the configuration and get the updated record
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  const record = {
+    id: database.hash(config.id),
+    controller: database.hash(controller),
+    meta,
+    config
+  };
+  try {
+    const result = await database.collections.dataHubConfig.insert(
+      record, database.writeOptions);
+    return result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate data hub configuration.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+};
+
+/**
+ * Retrieves all data hub configs matching the given query.
+ *
+ * @param {Object} actor the actor or capabilities for performing the action.
+ * @param {String} controller the controller for the data hubs to retrieve.
+ * @param {Object} query the optional query to use (default: {}).
+ * @param {Object} fields optional fields to include or exclude (default: {}).
+ * @param {Object} options options (eg: 'sort', 'limit').
+ *
+ * @return {Promise<Array>} resolves to the records that matched the query.
+ */
+api.findConfig = async (
+  {actor, controller, query = {}, fields = {}, options = {}}) => {
+  // check permission against any data hub matching the controller
+  await brPermissionCheck(
+    actor, PERMISSIONS.DATA_HUB_CONFIG_ACCESS,
+    {resource: [controller]});
+
+  // force controller ID
+  query.controller = database.hash(controller);
+  return database.collections.dataHubConfig.find(
+    query, fields, options).toArray();
+};
+
+// TODO: implement `updateConfig`
+
+/**
+ * Gets a data hub configuration.
+ *
+ * @param {Object} actor the actor or capabilities for performing the action.
+ * @param {String} id the ID of the data hub.
+ *
+ * @return {Promise<Object>} resolves to `{config, meta}`.
+ */
+api.getConfig = async ({actor, id}) => {
+  assert.string(id, 'id');
+
+  const record = await database.collections.dataHubConfig.findOne(
+    {id: database.hash(id)},
+    {_id: 0, config: 1, meta: 1});
+  if(!record) {
+    throw new BedrockError(
+      'Data hub configuration not found.',
+      'NotFoundError',
+      {dataHub: id, httpStatusCode: 404, public: true});
+  }
+
+  // check permission against data hub directly or its controller
+  await brPermissionCheck(
+    actor, PERMISSIONS.DATA_HUB_CONFIG_ACCESS,
+    {resource: [record.config.id, record.config.controller]});
+
+  return record;
+};
+
+/**
+ * Inserts a data hub document.
+ *
+ * @param {Object} actor the actor or capabilities for performing the action.
+ * @param {dataHubId} id the ID of the data hub to store the document in.
```

Comment:
Match the other uses:
```suggestion
 * @param {String} id the ID of the data hub to store the document in.
```